### PR TITLE
Print RAM warning message in online and offline modes

### PR DESCRIPTION
**Pull request by talonchandler**
This PR prints the following message to users who request a reconstruction (via online or offline modes) when <32 GB of RAM is available:

"WARNING: recOrder reconstructions often require more than the 16 GB of RAM that this computer is equipped with. We recommend starting with reconstructions of small volumes ~1000 x 1000 x 10 and working up to larger volumes while monitoring your RAM usage with Task Manager or htop."

When >32 GB is available (e.g. 64 GB), it prints:

"64 GB of RAM is available."

I'm happy to take wording edits or completely overhaul this PR given our discussion in #108. 

Note: `psutil` is a dependency of one of our dependencies, so it is already installed in `recOrder` environments.

### CODE REVIEW

**Comment by talonchandler** [comment]
This is ready to merge from my perspective. 

@mattersoflight does this adequately address #108? I don't think there's much need to use a more complete/complex solution since we're saving to disk before reconstruction.

File: recOrder/io/utils.py [hunk 1]
```diff
@@ -165,3 +167,16 @@ def get_unimodal_threshold(input_image):
             max_dist = per_dist
     assert best_threshold > -np.inf, 'Error in unimodal thresholding'
     return best_threshold
+
+def ram_message():
+    BYTES_PER_GB = 2**30
+    gb_available = psutil.virtual_memory().total / BYTES_PER_GB
+    if gb_available < 32:
+        return ' \n'.join(textwrap.wrap(
+               f'\033[91mWARNING:\033[0m recOrder reconstructions often require more than the {gb_available:.1f} ' \
```
Comment (ziw-liu) [inline recOrder/io/utils.py:176]: I'm not sure if we ever use `logging` in the CLI. If we do maybe it would make sense to not manually format the message?

Comment (ziw-liu) [reply recOrder/io/utils.py:176]: Also ANSI code support on Windows seems to be version-specific.

Comment (talonchandler) [reply recOrder/io/utils.py:176]: Good call on `print` vs. `logging`. I just changed the online mode `print`s to `logging.info`. 

Re: ANSI code support: I tested that it works in powershell, but you might be right with cmd.exe. I think the worst that will happen is that you'll see "\033[91mWARNING:\033[0m", which isn't the end of the world, so I'm not too concerned.

**Review by ziw-liu — APPROVED** [review]

File: recOrder/io/utils.py (hunk 1 above)
Comment (ziw-liu) [reply recOrder/io/utils.py:176]: We will see something like `[Warning] WARNING ...` in online mode, but that's only cosmetic.

Comment (ziw-liu) [reply recOrder/io/utils.py:176]: I did not think it through, we are now actually seeing:
```
10:21:26 INFO WARNING: recOrder reconstructions often require more than the
16.0 GB of RAM that this computer is equipped with. We recommend
starting with reconstructions of small volumes ~1000 x 1000 x 10 and
working up to larger volumes while monitoring your RAM usage with Task
Manager or htop.
```
And this message will only be displayed in console/log, without flashing a warning in the GUI.
Maybe we should have it do `show_warning(ram_message())` instead.

Comment (ziw-liu) [reply recOrder/io/utils.py:176]: Addressed in #174

**Pull request merged**